test/test_model.py

```diff
@@ -175,6 +175,18 @@ def test_config(self):
             # Confirm that we cannot modify config values.
             self.model.config['foo'] = 'bar'
 
+    def test_pod_spec(self):
+        os.environ['JUJU_UNIT_NAME'] = 'myapp/0'
+        self.addCleanup(os.environ.pop, 'JUJU_UNIT_NAME')
+        model = op.model.Model('myapp/0', ['db0', 'db1', 'db2'], op.model.ModelBackend())
+
+        fake_script(self, 'pod-spec-set', """cat - > $(dirname $0)/pod_spec.json""")
+        fake_script(self, 'pod-spec-get', """cat $(dirname $0)/pod_spec.json || echo {}""")
+
+        self.assertEqual(model.pod_spec, {})
+        model.pod_spec = {'foo': 'bar'}
+        self.assertEqual(model.pod_spec, {'foo': 'bar'})
```

Comment:
is there anything that we want to test the pod spec is cached from set and not read again?
(eg, change pod_spec.json behind its back and then assert that model.pod_spec hasn't changed?)